src/gameboy/PPU/Sprite.h

```diff
@@ -27,18 +21,66 @@ class Sprite {
 public:
     Sprite(uint8_t y, uint8_t x, uint8_t tileIndex, uint8_t flags, uint8_t positionInOAM);
 
-    bool coversLine(uint8_t line, unsigned int objectSize) const;
-
+    /**
+     * @return the x-coordinate of the sprite's top left corner.
+     */
     int getX() const;
+    /**
+     * @return the y-coordinate of the sprite's top left corner.
+     */
     int getY() const;
-    uint8_t getTileID(uint8_t lcdY, uint8_t objectSize) const;
+    /**
+     * Given a x-coordinate, determines which x-coordinate of the sprite it intersects.
+     * @param lcdX a given x-coordinate.
+     * @return the x-coordinate of the sprite that intersects lcdX.
+     */
     uint8_t getTileX(uint8_t lcdX) const;
+    /**
+     * Given a y-coordinate, determines which y-coordinate of the sprite it intersects.
+     * @param lcdY a given y-coordinate.
+     * @return the y-coordinate of the sprite that intersects lcdY.
+     */
     uint8_t getTileY(uint8_t lcdY) const;
+    /**
+     * Given a y-coordinate and the object size, determines the tile ID of the tile that intersects that y-coordinate.
+     * @param lcdY a given y-coordinate.
+     * @param objectSize a value that is 0 if sprites are composed of one tile and 1 if it is composed of two tiles.
+     * @return the tile ID of the tile that intersects the y-coordinate.
+     */
+    uint8_t getTileID(uint8_t lcdY, uint8_t objectSize) const;
+    /**
+     * @return the palette number associated with the sprite.
+     */
     uint8_t getPaletteNumber() const;
-    bool backgroundOverSprite() const;
-
-    void print() const;
+    /**
+     * @return the sprite's drawing priority over the background.
```

Comment:
`@return` should be at the bottom of the comment

Reply:
Done.